src/cargo/util/toml/mod.rs

```diff
@@ -3167,6 +3178,122 @@ impl<'de> de::Deserialize<'de> for TomlDebugInfo {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash, Serialize)]
+#[serde(untagged, rename_all = "kebab-case")]
+pub enum TomlTrimPaths {
+    Values(Vec<TomlTrimPathsValue>),
+    All,
+}
+
+impl<'de> de::Deserialize<'de> for TomlTrimPaths {
+    fn deserialize<D>(d: D) -> Result<TomlTrimPaths, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        use serde::de::Error as _;
+        let expecting = r#"a boolean, "none", "diagnostics", "macro", "object", "all", or an array with these options"#;
+        UntaggedEnumVisitor::new()
+            .expecting(expecting)
+            .bool(|value| {
+                Ok(if value {
+                    TomlTrimPaths::All
+                } else {
+                    TomlTrimPaths::none()
+                })
+            })
+            .string(|v| match v {
+                "none" => Ok(TomlTrimPaths::none()),
+                "all" => Ok(TomlTrimPaths::All),
+                v => {
+                    let d = v.into_deserializer();
+                    let err = |_: D::Error| {
+                        serde_untagged::de::Error::custom(format!("expected {expecting}"))
+                    };
+                    TomlTrimPathsValue::deserialize(d)
+                        .map_err(err)
+                        .map(|v| v.into())
+                }
+            })
+            .seq(|seq| {
+                let seq: Vec<String> = seq.deserialize()?;
+                let seq: Vec<_> = seq
+                    .into_iter()
+                    .map(|s| TomlTrimPathsValue::deserialize(s.into_deserializer()))
+                    .collect::<Result<_, _>>()?;
+                Ok(seq.into())
+            })
+            .deserialize(d)
+    }
+}
+
+impl TomlTrimPaths {
```

Comment:
nit: when i reorganized this file, I put all non-trait impl blocks first

Reply:
Oh how I missed that. Fixed!

is there any rustfmt or clippy rules to enforce this?